modules/apps/object/object-service/src/main/java/com/liferay/object/internal/model/listener/ObjectEntryModelListener.java

```diff
@@ -80,48 +88,79 @@ private void _executeObjectActions(
 				objectEntry.getModelClassName(), objectEntry.getCompanyId(),
 				objectActionTriggerKey,
 				_getPayloadJSONObject(
-					objectActionTriggerKey, originalObjectEntry, objectEntry),
+					objectActionTriggerKey, originalObjectEntry, objectEntry,
+					userId),
 				userId);
 		}
 		catch (PortalException portalException) {
 			throw new ModelListenerException(portalException);
 		}
 	}
 
+	private String _getExternalModel(ObjectEntry objectEntry, long userId)
+		throws PortalException {
+
+		User user = _userLocalService.getUser(userId);
+
+		DefaultDTOConverterContext defaultDTOConverterContext =
+			new DefaultDTOConverterContext(
+				false, Collections.emptyMap(), _dtoConverterRegistry, null,
+				user.getLocale(), null, user);
+
+		DTOConverter<ObjectEntry, ?> dtoConverter =
+			(DTOConverter<ObjectEntry, ?>)_dtoConverterRegistry.getDTOConverter(
+				ObjectEntry.class.getName());
+
+		try {
+			Object externalModel = dtoConverter.toDTO(
```

Comment:
Although it will throw a NullPointerException I will add an explicit condition to check if the dtoConverter is null and return the internal model

Reply:
Before this try-catch block I mean

Reply:
Done!